paddle/fluid/framework/data_feed.cc

```diff
@@ -35,9 +35,50 @@ limitations under the License. */
 #include "paddle/fluid/framework/feed_fetch_type.h"
 #include "paddle/fluid/platform/timer.h"
 
+
 namespace paddle {
 namespace framework {
 
+void RecordCandidateList::ReSize(size_t length) {
+    _mutex.lock();
+    _capacity = length;
+    CHECK(_capacity > 0);
+    _candidate_list.clear();
+    _candidate_list.resize(_capacity);
+    _full = false;
+    _cur_size = 0;
+    _total_size = 0;
+    _mutex.unlock();
+}
+
+void RecordCandidateList::ReInit() {
+    _mutex.lock();
+    _full = false;
+    _cur_size = 0;
+    _total_size = 0;
+    _mutex.unlock();
+}
+
+void RecordCandidateList::AddAndGet(const Record& record, RecordCandidate& result, std::shared_ptr<FleetWrapper>& fleet_ptr) {
```

Comment:
no need to pass "fleet_ptr" here ，you can use "auto fleet_ptr = FleetWrapper::GetInstance();"

Reply:
fixed